### DISCUSSION TITLE
Unable to use CUDA after WSL2 upgraded to Windows Store version

### DISCUSSION THREAD

**Discussion by ruicatxiao** [microsoft/terminal#14426] [Q&A] [answered]
I know this is likely a weird problem. Recently WSL2 prompt me to upgrade itself to a Windows Store based version. Before that I always used Windows Terminal to access my WSL2-Ubuntu setup. It worked flawlessly, able to use CUDA for everything. Then yesterday after WSL2 update to Windows Store version, all of a sudden I can no longer use any CUDA based apps through Windows terminal. 

The errorr message I have is 

NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.
Failed to properly shut down NVML: Driver Not Loaded

However, if I start WSL2-Ubuntu directly through the wsl.exe file, CUDA and everything works perfectly. What could be going on in Windows Terminal that prevents CUDA from being used? Any ideas?

**Comment by ruicatxiao** [comment, marked as answer]
Nvm I solved this by installing Windows Terminal from preview channel. Everything works fine now

**Reply by DHowett** [reply]
This is very concerning because Windows Terminal does not control access to CUDA.

**Reply by ruicatxiao** [reply]
Not sure what changed between Terminal and Terminal Preview. At least Terminal Preview work flawlessly for CUDA applications

**Reply by zadjii-msft** [reply]
The only _possible_ thing I can think of would be the addition of the Atlas engine in 1.16 Preview. Just to sanity check ourselves, could you try disabling that on the Preview build, (restart the Terminal) and check if the original issue repro's again? 

![image](https://user-images.githubusercontent.com/18356694/203399407-eea93004-8069-425c-ac15-312cbd2b57db.png)


That'll be the new renderer going forward, but I'd love to have this knowledge in my back pocket if it ever comes up again. Thanks!

**Reply by ruicatxiao** [reply]
I disabled it and did a full WSL shutdown through powershell. Restarted WSL via terminal, CUDA still works. For the sake of it I redownloaded Terminal from non-preview channel. And CUDA is not accessible from that way. This is very weird
